axum-macros/tests/from_request/fail/state_infer_multiple_different_types.stderr

```diff
@@ -0,0 +1,23 @@
+error[E0277]: the trait bound `AppState: FromRef<S>` is not satisfied
+ --> tests/from_request/fail/state_infer_multiple_different_types.rs:6:18
+  |
+6 |     inner_state: State<AppState>,
+  |                  ^^^^^ the trait `FromRef<S>` is not implemented for `AppState`
+  |
+  = note: required because of the requirements on the impl of `FromRequestParts<S>` for `State<AppState>`
+help: consider extending the `where` clause, but there might be an alternative better way to express this requirement
+  |
+4 | #[derive(FromRequest, AppState: FromRef<S>)]
+  |                     ++++++++++++++++++++++
+
+error[E0277]: the trait bound `OtherState: FromRef<S>` is not satisfied
+ --> tests/from_request/fail/state_infer_multiple_different_types.rs:7:18
+  |
+7 |     other_state: State<OtherState>,
+  |                  ^^^^^ the trait `FromRef<S>` is not implemented for `OtherState`
+  |
+  = note: required because of the requirements on the impl of `FromRequestParts<S>` for `State<OtherState>`
+help: consider extending the `where` clause, but there might be an alternative better way to express this requirement
+  |
+4 | #[derive(FromRequest, OtherState: FromRef<S>)]
+  |                     ++++++++++++++++++++++++
```

Comment:
I think there's room for improvement on this error message, but the important bit is that we have a test for it not working. I can work on the error message after this is merged.